src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -407,30 +497,20 @@ def launch_fluent(
             )
         )
     if start_instance:
-        if FLUENT_EXE_PATH:
-            exe_path = FLUENT_EXE_PATH[0]
-        else:
-            exe_path = _get_fluent_exe_path()
+        exe_path = _get_fluent_exe_path()
         launch_string = exe_path
         launch_string += _build_fluent_launch_args_string(**argvals)
         if meshing_mode:
             launch_string += " -meshing"
         server_info_filepath = _get_server_info_filepath()
+
         try:
             launch_string += f" {additional_arguments}"
             launch_string += f' -sifile="{server_info_filepath}"'
             launch_string += " -nm"
-            if (show_gui is False) or (
-                show_gui is None and (os.getenv("PYFLUENT_SHOW_SERVER_GUI") != "1")
-            ):
-                aargs = additional_arguments
-                if (
-                    "-g " not in aargs
-                    and "-gu " not in aargs
-                    and not aargs.endswith("-g")
-                    and not aargs.endswith("-gu")
-                ):
-                    launch_string += " -hidden"
+            launch_string = _update_launch_string_wrt_gui_options(
```

Comment:
@prmukherj can call one top-level function to update the launch string. That can delegate to `_update_launch_string_wrt_gui_options` but also execute other code before that (i.e. the above steps).

Reply:
@seanpearsonuk, I am still thinking o this :)